File: backend/src/main/java/com/gymapp/modules/training/dto/WorkoutPlanRequest.java
package com.gymapp.modules.training.dto;

import com.gymapp.modules.training.enums.WpType;
import com.gymapp.modules.training.enums.TargetLevel;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;
import java.util.List;
import java.util.UUID;

@Data
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class WorkoutPlanRequest {
    @NotBlank(message = "Name is required")
    private String name;
    private String description;
    @NotNull(message = "Plan type is required")
    private WpType planType;
    private TargetLevel targetLevel;
    private UUID assignedTo;
    private List<PlanExerciseRequest> exercises;
    private String scheduledDate;
}

File: android/app/src/main/java/com/gymapp/android/data/remote/dto/WorkoutPlanRequest.kt
package com.gymapp.android.data.remote.dto

data class WorkoutPlanRequest(
    val name: String,
    val description: String?,
    val planType: String,
    val targetLevel: String?,
    val assignedTo: String? = null,
    val scheduledDate: String? = null,  //
    val exercises: List<PlanExerciseRequest>
)